runtime/lua/vim/lsp/snippet.lua

```diff
@@ -0,0 +1,7 @@
+local parser = require('vim.lsp.snippet.parser')
+local ast = require('vim.lsp.snippet.ast')
+
+return {
+  parser = parser,
```

Comment:
No need for this. Just put the parse() function directly in this snippet.lua module. The module tree should not have so much depth unless it is truly needed.

It does look like `snippet/ast.lua` module is justified, because its functions operate on a shared datatype.